frame/election-providers/src/two_phase/macros.rs

```diff
@@ -0,0 +1,95 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Some helper macros for this crate.
+
+#[macro_export]
+macro_rules! log {
+	($level:tt, $patter:expr $(, $values:expr)* $(,)?) => {
+		frame_support::debug::$level!(
+			target: $crate::LOG_TARGET,
+			concat!("🏦 ", $patter) $(, $values)*
+		)
+	};
+}
+
+#[macro_export]
+macro_rules! voter_index_fn {
+	($voters:ident, $acc:ty, $t:ident) => {
+		|who: &$acc| -> Option<$crate::two_phase::CompactVoterIndexOf<$t>> {
+			$voters
+				.iter()
+				.position(|(x, _, _)| x == who)
+				.and_then(|i| <usize as $crate::TryInto<$crate::two_phase::CompactVoterIndexOf<$t>>>::try_into(i).ok())
+			}
+	};
+}
+
+#[macro_export]
+macro_rules! target_index_fn {
+	($targets:ident, $acc:ty, $t:ident) => {
+		|who: &$acc| -> Option<$crate::two_phase::CompactTargetIndexOf<$t>> {
+			$targets
+				.iter()
+				.position(|x| x == who)
+				.and_then(|i| <usize as $crate::TryInto<$crate::two_phase::CompactTargetIndexOf<$t>>>::try_into(i).ok())
+			}
+	};
+}
+
+#[macro_export]
+macro_rules! voter_at_fn {
+	($snap:ident, $acc:ty, $t:ident) => {
+		|i: $crate::two_phase::CompactVoterIndexOf<$t>| -> Option<$acc> {
+							<$crate::two_phase::CompactVoterIndexOf<$t> as $crate::TryInto<usize>>::try_into(i)
+								.ok()
+								.and_then(|i| $snap
+									.get(i)
+									.map(|(x, _, _)| x)
+									.cloned()
+								)
+							}
+	};
+}
+
+#[macro_export]
+macro_rules! target_at_fn {
```

Comment:
Maybe I am missing something, but couldnt these just be normal functions?

Reply:
I also don't see a reason now to favour either of them, except that macros might be a bit faster as you don't need to move data around, and, not deal with verbose function signatures. 

Both work in principle though. I would keep it as-is.

Reply:
np with me. I bet some other rustaceans might have issue with you though 😈

Reply:
note that you can flag `#[inline]` if you don't want to move data around

Reply:
So I actually tried this. I made an inline function to get voter index, which is used the most in feasibility check, tried benchamrking on my own machine, wasm compiler, 20 steps, 10 reps, and the coefficient of `a` (active voters) in both cases is:

- function: 30.85
- macro: 27

I repeated all results twice, so I guess there's a small difference

Either way this can be an easy issue to pickup later on.